guide/api-design-guidelines.md

```diff
@@ -0,0 +1,181 @@
+The secrets are stored in secret store resource [Applications.Core/secretStores](https://docs.radapp.io/reference/resource-schema/core-schema/secretstore/)
+
+Our TypeSpec definitions provide the `SecretReference` and `SecretConfig` model types to standardize usage. These are prescribed to ensure sensitive information is handled securely and consistently across different components and resources in Radius. The following are their definitions along with usage examples:
+
+### SecretReference Model
+
+<a href="#secret-model" name="secret-model">:white_check_mark:</a> **DO** follow this structure when adding support for secrets to resources or components in Radius.
+
+```
+
+@doc("This specifies a reference to a secret. Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+model SecretReference {
+  @doc("The ID of an Applications.Core/SecretStore resource containing sensitive data.")
+  source: string;
+
+  @doc("The key for the secret in the secret store.")
+  key: string;
+}
+
+```        
+
+#### Usage of SecretReference
+
+<a href="#secret-envvar" name="secret-envvar">:white_check_mark:</a> **DO** use the `SecretReference` type to reference a single (scalar) secret value in a resource property. Resource properties should reference a `Applications.Core/secretStores` instead of directly containing secret data.
+
+This pattern simplifies the overall design of Radius by reducing the number of places where we store secret data. 
+This solution is in alignment with Kubernetes design patterns. The structure also allows for environment variables to refer to other resources such as ConfigMaps, Pod Fields etc. in the future. 
+Examples include cases where environment variables containing sensitive information are injected into a container or used in a Terraform execution process.
+
+#### Example TypeSpec Definition
+
+```
+
+@doc("environment")
+env?: Record<EnvironmentVariable>;
+
+@doc("Environment variables type")
+model EnvironmentVariable {
+
+  @doc("The value of the environment variable")
+  value?: string;
+
+  @doc("The reference to the variable")
+  valueFrom?: EnvironmentVariableReference;
+}
+
+@doc("The reference to the variable")
+model EnvironmentVariableReference {
+  @doc("The secret reference")
+  secretRef: SecretReference;
+}
+
+```
+
+#### Example Bicep Definition
+
+```bicep
+
+env: {
+  DB_USER: { value: 'DB_USER' }
+  DB_PASSWORD: {
+    valueFrom: {
+      secretRef: {
+        source: secret.id
+        key: 'DB_PASSWORD'
+      }
+    } 
+  }
+} 
+
+```
+
+### SecretConfig Model
+
+<a href="#secretconfig-model" name="secretconfig-model">:white_check_mark:</a> **DO** follow this structure when a component or resource in Radius requires authentication to external systems, such as private container registries, TLS certificates.
```

Comment:
i think we can rephrase this into something like : "follow this structure when required keys are known"? e.g  secret store used for private terraform recipe config, we know the expected secret keys are `username` and `pat`